### PR TITLE
Use workerqueue for mailstream jobs instead of custom table

### PR DESCRIPTION
**Pull request by mexon**
This uses a completely different approach to managing retries, that avoids the need to create a whole table just for this plugin.

The main reason I need retries is to work nicely with retriever (which I will eventually submit, but I still need to make it use the workerqueue).  Retriever marks items as invisible until it's finished processing them, and they shouldn't be mailed until then.  Mailstream will now retry every hour until it succeeds.  I'm hoping this doesn't cause too many problems.

### CODE REVIEW

File: mailstream/mailstream.php [hunk 1]
```diff
@@ -114,16 +105,40 @@ function mailstream_generate_id($uri)
...
+	$user = User::getById($item['uid']);
+	if (empty($user)) {
+			Logger::error('mailstream_send_hook could not fund user', ['uid' => $item['uid']]);
+		return;
+	}
+
+	if (!mailstream_send($data['message_id'], $item, $user)) {
+		$delayed = date(DateTimeFormat::utc('now + 1 hour'));
+		$data['tries'] += 1;
+		Hook::fork(['priority' => PRIORITY_LOW, 'delayed' => $delayed], 'mailstream_send_hook', $data);
```
Comment (annando) [inline mailstream/mailstream.php]: Please have a look here: https://github.com/friendica/friendica/blob/2021.06-rc/src/Worker/APDelivery.php#L104

I guess that `Worker::defer()` can help here. it automatically calculates the time between retrials, so you don't need to have a look at that as well.

**Comment by annando** [comment]
Have you tested it on your system?

**Comment by mexon** [comment]
> Have you tested it on your system?

I'm running it on my test system, with a patch that makes mail sending randomly fail 7/8 of the time to properly exercise the code.  The test system generates a post every 20 minutes, and I'm watching the emails arrive as well as watching the retries in the workerqueue.  It seems pretty robust, all the emails do arrive eventually.

I haven't put it on my live system yet.  The fun part is the conversion of the existing custom table when you upgrade.  Before I upgrade my live system I'd have to make a site backup, and since the database is quite large I've been avoiding that.  Would you like me to run it on live as well before accepting the PR?

I'm not quite sure why the build above is failing?

**Comment by annando** [comment]
@nupplaphil do you know why it fails?

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**